packages/rum-core/src/domain/startCustomerDataTelemetry.spec.ts

```diff
@@ -0,0 +1,165 @@
+    batchBytesCount?: number
+    contextBytesCount?: number
+    context?: Context
+  }) {
+    fakeContextBytesCount = contextBytesCount
+    fakeContext = context
+
+    for (let index = 0; index < eventNumber; index++) {
+      lifeCycle.notify(LifeCycleEventType.RUM_EVENT_COLLECTED, { type: RumEventType.VIEW } as RumEvent & Context)
+    }
+    batchFlushObservable.notify({ bufferBytesCount: batchBytesCount, bufferMessagesCount: eventNumber })
+  }
+
+  function spyOnContextManager(contextManager: ContextManager) {
+    spyOn(contextManager, 'get').and.callFake(() => fakeContext)
+    spyOn(contextManager, 'getBytesCount').and.callFake(() => fakeContextBytesCount)
+  }
+
+  function spyOnFeatureFlagContexts(featureFlagContexts: FeatureFlagContexts) {
+    spyOn(featureFlagContexts, 'findFeatureFlagEvaluations').and.callFake(() => fakeContext)
+    spyOn(featureFlagContexts, 'getFeatureFlagBytesCount').and.callFake(() => fakeContextBytesCount)
+  }
+
+  beforeEach(() => {
+    updateExperimentalFeatures(['customer_data_telemetry'])
+    setupBuilder = setup()
+      .withFakeClock()
+      .withConfiguration({
+        telemetrySampleRate: 100,
+        customerDataTelemetrySampleRate: 100,
+        maxTelemetryEventsPerPage: 2,
+      })
+      .beforeBuild(({ globalContextManager, userContextManager, featureFlagContexts, configuration }) => {
+        batchFlushObservable = new Observable()
+        lifeCycle = new LifeCycle()
+        fakeContextBytesCount = 1
+        fakeContext = { foo: 'bar' }
+        spyOnContextManager(globalContextManager)
+        spyOnContextManager(userContextManager)
+        spyOnFeatureFlagContexts(featureFlagContexts)
+
+        telemetryEvents = []
+        const telemetry = startTelemetry(TelemetryService.RUM, configuration)
+        telemetry.observable.subscribe((telemetryEvent) => telemetryEvents.push(telemetryEvent))
+
+        startCustomerDataTelemetry(
+          configuration,
+          telemetry,
+          lifeCycle,
+          globalContextManager,
+          userContextManager,
+          featureFlagContexts,
+          batchFlushObservable
+        )
+      })
+  })
+
+  afterEach(() => {
+    setupBuilder.cleanup()
+    resetExperimentalFeatures()
+  })
+
+  it('should collect customer data telemetry', () => {
+    const { clock } = setupBuilder.build()
+
+    generateBatch({ eventNumber: 2, contextBytesCount: 2, batchBytesCount: 2 })
+    generateBatch({ eventNumber: 1, contextBytesCount: 1, batchBytesCount: 1 })
+    clock.tick(MEASURES_PERIOD_DURATION)
+
+    expect(telemetryEvents[0].telemetry).toEqual({
+      type: 'log',
+      status: 'debug',
+      message: 'Customer data measures',
+      batchCount: 2,
+      batchBytesCount: { min: 1, max: 2, sum: 3 },
+      batchMessagesCount: { min: 1, max: 2, sum: 3 },
+      globalContextBytes: { min: 1, max: 2, sum: 5 },
+      userContextBytes: { min: 1, max: 2, sum: 5 },
+      featureFlagBytes: { min: 1, max: 2, sum: 5 },
+    })
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏ ‏it may be easier to guess what is going on by using different magnitudes
```suggestion
    generateBatch({ eventNumber: 10, contextBytesCount: 10, batchBytesCount: 10 })
    generateBatch({ eventNumber: 1, contextBytesCount: 1, batchBytesCount: 1 })
    clock.tick(MEASURES_PERIOD_DURATION)

    expect(telemetryEvents[0].telemetry).toEqual({
      type: 'log',
      status: 'debug',
      message: 'Customer data measures',
      batchCount: 2,
      batchBytesCount: { min: 1, max: 10, sum: 11 },
      batchMessagesCount: { min: 1, max: 10, sum: 11 },
      globalContextBytes: { min: 1, max: 10, sum: 101 },
      userContextBytes: { min: 1, max: 10, sum: 101 },
      featureFlagBytes: { min: 1, max: 10, sum: 101 },
    })
```